sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdClientChannelPool.java

```diff
@@ -661,6 +707,7 @@ private double computeLoadFactor() {
     }
 
     private void doAcquireChannel(final Promise<Channel> promise, final Channel candidate) {
+        acquiredChannels.put(candidate, candidate);
```

Comment:
Why is it pulled out of EventLoop ?

Reply:
doChannelHealthChecks interacts with the Channel and needs to be executed on the channel's eventLoop thread. This usually is a different thread than the channel pool's event loop thread - technically could be the same thought as well.
For clarity added a this.ensureInEventLoop (which happened already before - so just to make ti more robust)